src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -333,6 +333,110 @@ public void TestAddingRelationshipWithDisabledGraphQL()
+        [DataTestMethod]
+        [DataRow(true, DisplayName = "Validate that exception is thrown when GQL is enabled and user defines multiple relationship between source and target entities.")]
+        [DataRow(false, DisplayName = "Validate that no exception is thrown when GQL is disabled and user defines multiple relationship between source and target entities.")]
+        public void TestMultipleRelationshipsBetweenSourceAndTargetEntities(bool isGQLEnabledForSource)
+        {
+            string sourceEntityName = "SourceEntity", targetEntityName = "TargetEntity";
+
+            // Create relationship between source and target entities.
+            EntityRelationship relationship = new(
+                Cardinality: Cardinality.One,
+                TargetEntity: targetEntityName,
+                SourceFields: new string[]{ "abc" },
+                TargetFields: new string[] { "xyz" },
+                LinkingObject: null,
+                LinkingSourceFields: null,
+                LinkingTargetFields: null
+            );
+
+            // Add another relationship between the same source and target entities.
+            EntityRelationship duplicateRelationship = new(
+                Cardinality: Cardinality.Many,
+                TargetEntity: targetEntityName,
+                SourceFields: null,
+                TargetFields: null,
+                LinkingObject: null,
+                LinkingSourceFields: null,
+                LinkingTargetFields: null
+            );
+
+            string relationshipName = "relationship", duplicateRelationshipName = "duplicateRelationship";
+            Dictionary<string, EntityRelationship> relationshipMap = new()
+            {
+                { relationshipName, relationship },
+                { duplicateRelationshipName, duplicateRelationship }
+            };
+
+            // Creating source entity with enabled graphQL
+            Entity sourceEntity = GetSampleEntityUsingSourceAndRelationshipMap(
+                source: "TEST_SOURCE1",
+                relationshipMap: relationshipMap,
+                graphQLDetails: new(Singular: "", Plural: "", Enabled: isGQLEnabledForSource)
+            );
+
+            // Creating target entity.
+            Entity targetEntity = GetSampleEntityUsingSourceAndRelationshipMap(
+                source: "TEST_SOURCE2",
+                relationshipMap: null,
+                graphQLDetails: new("", "", true)
+            );
+
+            Dictionary<string, Entity> entityMap = new()
+            {
+                { sourceEntityName, sourceEntity },
+                { targetEntityName, targetEntity }
+            };
+
+            RuntimeConfig runtimeConfig = new(
+                Schema: "UnitTestSchema",
+                DataSource: new DataSource(DatabaseType: DatabaseType.MSSQL, "", Options: null),
+                Runtime: new(
+                    Rest: new(),
+                    GraphQL: new(),
+                    Host: new(null, null)
```

Comment:
how do you ensure `development` mode is picked up? Is that the default today?

Reply:
The default is production mode. However, the check for the host mode is done in startup:

<img width="626" alt="image" src="https://github.com/Azure/data-api-builder/assets/34566234/57409a9c-e39c-4924-87f9-d9906b19d85e">

So, we are unit testing the method for which it does not really matter what is the mode- since the method does not have a dependency on the mode. We can make mode check a part of this method (which I think is something we should do- would make the method more complete and encapsulate all the logic together). Let me know what your thoughts are.